src/modules/fancyzones/lib/KeyState.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "GenericKeyHook.h"
+
+template<int... keys>
+class KeyState
+{
+public:
+    KeyState(const std::function<void()>& callback) :
+        m_state(false),
+        m_hook(std::bind(&KeyState::onChangeState, this, std::placeholders::_1)),
+        m_updateCallback(callback)
+    {
+    
+    }
+
+    inline bool state() const noexcept { return m_state; }
+
+    inline void enable()
+    {
+        m_hook.enable();
+        m_state = (((GetAsyncKeyState(keys) & 0x8000) || ...));
+    }
+
+    inline void disable()
+    {
+        m_hook.disable();
+    }
+
+    inline void setState(bool state) noexcept
+    {
+        if (m_state != state)
+        {
+            m_state = state;
+            if (m_updateCallback)
+            {
+                m_updateCallback();
+            }
+        }
+    }
+
+private:
+    inline void onChangeState(bool state) noexcept
+    {
+        setState(state);
+    }
+
+private:
+    std::atomic<bool> m_state;
```

Comment:
nit: State could be represented as enum to make clear what state means. But as this is key state, it's ok